ast/compile.go

```diff
@@ -4663,6 +4663,17 @@ func rewriteDeclaredVarsInTerm(g *localVarGenerator, stack *localDeclaredVars, t
 			return true, errs
 		}
 		return false, errs
+	case Call:
+		ref := v[0]
+		WalkVars(ref, func(v Var) bool {
+			if gv, ok := stack.Declared(v); ok && !gv.Equal(v) {
+				// We will rewrite the ref of a function call, which is never ok
```

Comment:
...since we don't have first-class functions. So you cannot do
```rego

f(g) := x {
  x := g(100)
}
```
or some such thing.